homeassistant/components/huawei_lte/__init__.py

```diff
@@ -444,13 +438,8 @@ def _update_router(*_: Any) -> None:
         router.update()
 
     # Set up periodic update
-    router.unload_handlers.append(
-        async_track_time_interval(hass, _update_router, SCAN_INTERVAL)
-    )
-
-    # Clean up at end
     config_entry.async_on_unload(
-        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, router.cleanup)
+        async_track_time_interval(hass, _update_router, SCAN_INTERVAL)
```

Comment:
Is the router clean up on Home Assistant stop not needed any more?

Reply:
I'm assuming (have not done any real homework) that HA will unload config entries on stop. If that's correct, then a separate cleanup on stop is redundant. If that's incorrect, than I would fix this in HA :)

Reply:
We don't unload config entries on Home Assistant stop.

Reply:
Do you know why, though? Seems to me that unload on stop would be the right thing to do.

Reply:
We don't need to do all the work that the unload does when stopping Home Assistant. Eg unloading the platforms isn't needed. It's mostly network or serial connections that need to be closed.

Reply:
Fair enough, results in some duplicate code though. Will try to remember to address this in huawei_lte soon.

Reply:
#49871